File: backend/api/models/Service.php
<?php
class Service
{
    private $conn;
    private $table = "services";

    public $id;
    public $provider_id;
    public $title;
    public $description;
    public $category;
    public $location;
    public $contact_info;

    public function __construct($db)
    {
        $this->conn = $db;
    }

    // Create a service
    public function create()
    {
        $query = "INSERT INTO " . $this->table . " (provider_id, title, description, category, location, contact_info) 
                  VALUES (:provider_id, :title, :description, :category, :location, :contact_info)";
        $stmt = $this->conn->prepare($query);

        $stmt->bindParam(":provider_id", $this->provider_id);
        $stmt->bindParam(":title", $this->title);
        $stmt->bindParam(":description", $this->description);
        $stmt->bindParam(":category", $this->category);
        $stmt->bindParam(":location", $this->location);
        $stmt->bindParam(":contact_info", $this->contact_info);

        return $stmt->execute();
    }

    // Fetch all services
    public function readAll()
    {
        $query = "SELECT * FROM " . $this->table;
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        return $stmt;
    }

    // Fetch services by category
    public function readByCategory($category)
    {
        $query = "SELECT * FROM " . $this->table . " WHERE category = :category";
        $stmt = $this->conn->prepare($query);
        $stmt->bindParam(":category", $category);
        $stmt->execute();
        return $stmt;
    }
}

File: frontend/lib/models/service.dart
class Service {
  final int id;
  final int providerId;
  final String title;
  final String description;
  final String category;
  final String location;
  final String contactInfo;

  Service({
    required this.id,
    required this.providerId,
    required this.title,
    required this.description,
    required this.category,
    required this.location,
    required this.contactInfo,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id'],
      providerId: json['provider_id'],
      title: json['title'],
      description: json['description'],
      category: json['category'],
      location: json['location'],
      contactInfo: json['contact_info'],
    );
  }
}
